src/python/pants/option/option_types.py

```diff
@@ -14,6 +15,11 @@
 _ValueT = TypeVar("_ValueT")
 # NB: We don't provide constraints, as our `XListOption` types act like a set of contraints
 _ListMemberType = TypeVar("_ListMemberType")
+_HelpT = Union[Callable[[Any], str], str]
```

Comment:
Could this be `Callalable[type[Subsystem]]`? If MyPy doesn't like it, even commenting that would help this code to be more declarative. Right now I had to read around to figure out what the arg would be.